src/main/java/lotto/model/LottoNumbers.java

```diff
@@ -0,0 +1,56 @@
+package lotto.model;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import lotto.model.enums.MatchCount;
+
+public class LottoNumbers {
+    static final int NUMBER_SIZE = 6;
+    static final String NUMBER_SIZE_ERR_MSG = "로또 숫자는 서로 다른 " + NUMBER_SIZE + "개로 구성되어야 합니다.";
+
+    private final SortedSet<Number> numbers;
+
+    public LottoNumbers(int... numbers) {
+        this(() -> Arrays.stream(numbers).mapToObj(Number::ofValue).collect(Collectors.toSet()));
+    }
+
+    public LottoNumbers(NumberSupplier numberSupplier) {
```

Comment:
```suggestion
    public LottoNumbers(Set<Number> numbers) {
```
이렇게 처리해도 되지 않나요?
NumberSupplier 인터페이스가 필요한 이유가 있을까요? 너무 구조를 복잡하게 가져가는 것은 아닐까요?

Reply:
박재성님 강의에서 자동차경주게임 구현 시, 랜덤 넘버를 이런 식으로 주입하는 걸 본 기억이 있어서 이렇게 해봤습니다. 공부를 할 때는 좀 오버엔지니어링을 할 필요가 있다고 하셔서요^^;; 근데 저도 너무 복잡한 것 같긴 합니다. 수정하도록 하겠습니다!